overlord/snapstate/snapmgr.go

```diff
@@ -688,9 +733,12 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	cand := snapst.Candidate
-
 	m.backend.Candidate(snapst.Candidate)
-	snapst.Sequence = append(snapst.Sequence, snapst.Candidate)
+	hadCandidateInSeq := false
```

Comment:
this seems reversed, it true by default (and false if we had to change seq to add the cand there)